src/quo/components/wallet/transaction_progress/view.cljs

```diff
@@ -0,0 +1,189 @@
+(def ^:private max-sidenet-verifications 1)
+
+(defn- icon-internal
+  ([icon]
+   (icon-internal icon nil 20))
+  ([icon color]
+   (icon-internal icon color 20))
+  ([icon color size]
+   [rn/view {:style style/icon}
+    [icons/icon
+     icon
+     {:color    color
+      :size     size
+      :no-color (when (not color) true)}]]))
+
+(defn- text-internal
+  [{:keys [weight size accessibility-label color]
+    :or   {weight :semi-bold
+           size   :paragraph-1}}
+   title]
+  [text/text
+   {:accessibility-label accessibility-label
+    :ellipsize-mode      :tail
+    :number-of-lines     1
+    :weight              weight
+    :size                size
+    :style               {:color color}}
+   title])
+
+(defn- network-type-text
+  [state]
+  (cond
+    (or (= state :sending) (= state :pending))      (i18n/label :t/pending-on)
+    (or (= state :confirmed) (= state :finalising)) (i18n/label :t/confirmed-on)
+    (= state :finalized)                            (i18n/label :t/finalized-on)
+    (= state :error)                                (i18n/label :t/failed-on)))
+
+(defn- calculate-counter
+  [counter]
+  (if (< counter 4) counter max-mainnet-verifications))
+
+(defn mainnet-label
+  [counter]
+  (str counter "/" max-mainnet-verifications))
+
+(defn subnet-label
+  [counter]
+  (str counter "/" max-sidenet-verifications))
+
+(defn- text-steps
+  [network state epoch-number counter]
+  (let [steps (case network
+                :mainnet
+                {:pending    (mainnet-label 0)
+                 :sending    (mainnet-label (calculate-counter counter))
+                 :confirmed  (mainnet-label (calculate-counter counter))
+                 :finalising (mainnet-label (calculate-counter counter))
+                 :finalized  (i18n/label :t/epoch-number {:number epoch-number})
+                 :error      (mainnet-label 0)}
+                (or :optimism :arbitrum)
+                {:pending    (subnet-label 0)
+                 :sending    (subnet-label 0)
+                 :confirmed  (subnet-label 0)
+                 :finalising (subnet-label 1)
+                 :finalized  (i18n/label :t/epoch-number {:number epoch-number})
+                 :error      (subnet-label 0)}
+                nil)]
+    (get-in steps [state])))
+
+(defn- get-status-icon
+  [theme state]
+  (cond
+    (or (= state :pending)
+        (= state :sending))    [:i/pending-state
+                                (colors/theme-colors colors/neutral-50
+                                                     colors/neutral-40
+                                                     theme)]
+    (or (= state :confirmed)
+        (= state :finalising)) [:i/positive-state (colors/resolve-color :success theme)]
+    (= state :finalized)       [:i/diamond]
+    (= state :error)           [:i/negative-state (colors/resolve-color :danger theme)]))
+
+(defn- get-network
+  [networks network]
+  (first (filter #(= (:network %) network) networks)))
```

Comment:
@smohamedjavid @mmilad75 

I'd suggest instead of `filter` + `first`  we just use `some`. It returns the first logical true value.

```clojure
(defn- get-network
  [networks network]
  (some #(when (= (:network %) network) %)
        networks))
```

I know, the code is a little more verbose, but I think that's the purpose of `some`.

Reply:
Done